daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/NodeInfo.scala

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.transaction
+
+/** Trait for extracting information from an abstract node.
+  * Used for sharing the implementation of common computations
+  * over nodes and transactions.
+  */
+trait NodeInfo[PartyRep] {
+
+  /** The node kind, e.g. create, consuming exercise etc. */
+  def kind: NodeKind
+
+  /** The signatories of the contract associated with the node. */
+  def signatories: Set[PartyRep]
+
+  /** The stakeholders of the contract associated with the node. */
+  def stakeholders: Set[PartyRep]
+
+  /** The actors (e.g. the controlling parties of the choice) of the node. */
+  def actors: Set[PartyRep]
+}
+
+object NodeInfo {
+
+  /** Compute the informees of a node based on the ledger model definition.
+    *
+    * Refer to https://docs.daml.com/concepts/ledger-model/ledger-privacy.html#projections
+    */
+  def informeesOfNode[PartyRep](node: NodeInfo[PartyRep]): Set[PartyRep] =
+    node.kind match {
```

Comment:
Looks like `NodeKind` was only introduced to avoid a more verbose/noisy pattern match? This, coupled with a too wide definition of the interface `NodeInfo`, I'd rather err on the side of not introducing it.

Reply:
What would you suggest instead? This implementation is used on other “Node” implementations than the one defined by Node.scala.

Reply:
Why not push this inside of the trait? Especially since this is something that every trait implementation actually _can_ sensibly define?

Reply:
Done.